ci: add test summary using JUnitXML (#346)

Drranny opened the pull request:
Purpose:

- Add a FAILED-only test summary to CI to quickly spot failing tests.
- Resolves  [[#346](https://github.com/pyodide/pyodide-recipes/issues/346)](https://github.com/pyodide/pyodide-recipes/issues/346).

Changes:

- `build.yml`: add pytest JUnitXML output, integrate `pmeier/pytest-results-action@v0.7.2`, upload artifacts
- Use third-party action’s summary generator instead of custom parser
- Show overall stats and only failed or error tests (`display-options: fE`)
- Store per-job JUnitXML as artifacts (`test-results-chrome`, `test-results-firefox`, `test-results-node`)

Test plan:

- Verified in fork with 14 CI jobs across Chrome, Firefox, and Node
- Added dummy failing tests in several packages: numpy, pandas, scipy, matplotlib, geopandas
- Confirmed that JUnitXML files are generated for each matrix
- Confirmed that artifacts (`test-results-*.xml`) are uploaded correctly
- Confirmed that summary tables are visible in GitHub Actions UI (passed, skipped, failed, error)
- Confirmed that only failed and error tests are listed with inline messages
- Confirmed that results are consistent across all environments (see attached screenshot)

References:

- [`[pmeier/pytest-results-action](https://github.com/pmeier/pytest-results-action)`](https://github.com/pmeier/pytest-results-action) for summary generation
- [[pytest JUnitXML documentation](https://docs.pytest.org/en/stable/how-to/output.html#creating-junitxml-format-files)](https://docs.pytest.org/en/stable/how-to/output.html#creating-junitxml-format-files) for XML export configuration
- CircleCI JUnitXML viewer used as reference for target UI


<img width="909" height="800" alt="스크린샷 2025-10-04 오후 3 59 35" src="https://github.com/user-attachments/assets/7278119f-af61-445c-a998-6348412b8997" />
<img width="902" height="181" alt="스크린샷 2025-10-04 오후 4 00 04" src="https://github.com/user-attachments/assets/e540ba73-f875-4519-b707-c5e51c1c5926" />
<img width="873" height="796" alt="스크린샷 2025-10-04 오후 3 59 55" src="https://github.com/user-attachments/assets/699b2e75-bc4c-4d50-aa8f-1b95fb394f05" />
<img width="892" height="509" alt="스크린샷 2025-10-04 오후 3 59 47" src="https://github.com/user-attachments/assets/d26b4172-6ae6-4d33-a341-4698d283423f" />

ryanking13 reviewed (commented):
Thanks! I left some minor comments, otherwise looks good to me.

Drranny commented:
> Thanks! I left some minor comments, otherwise looks good to me.

Thanks for the detailed review!
I’ve applied all the suggested changes:
- Security: Replaced version tags with commit hashes for both pytest-results-action and actions/upload-artifact.
- Simplification: Removed the SciPy test summary since it’s not commonly used.

These updates keep the workflow consistent with the existing jobs.
Let me know if there’s anything else I should tweak!

ryanking13 reviewed (approved):
Thanks!
